contrib/babelfishpg_tsql/src/pl_comp.c

```diff
@@ -1642,6 +1667,203 @@ pltsql_param_ref(ParseState *pstate, ParamRef *pref)
 	return make_datum_param(expr, nse->itemno, pref->location);
 }
 
+/*
+ * pltsql_post_func_ref		parser callback for FuncRefs to check and modify Geospatial function call
+ */
+static List *
+pltsql_post_func_ref(ParseState *pstate, FuncCall *fn, List *fargs)
+{
+	List *ret = resolve_geospatial_func_ref(pstate, fn, fargs);
+	if (ret)
+	{
+		elog(WARNING, "got some rets %d", ret->length);
+		return ret;
+	}
+	return fargs;
+}
+
+/*
+ * This function reverts schema name, which is already rewritten by rewrite_column_refs or rewrite_plain_name functions
+ * This is required so that parser identifies the name for Geospatial Function call
+ */
+static char *
+resolve_schema_name(char *db_name, char *schema_name, MigrationMode mode)
```

Comment:
Eg: select t2.geom.STX from t2; (where 'geom' is geometry column in table t2)
 In this Case, 't2.geom.STX' will rewritten as 'master_t2.geom.STX'
 because t2 is considered as Schema name and not table name because of 3 part call. So, we have to resolve it